Fix 78 UI test failures from gateway setup running in jsdom

## Problem

When `.local/openclaw-gateway` exists, `vitest.config.ts` globally includes `.local/openclaw-gateway/test/setup.ts` as a setup file. This file uses relative imports (`../src/...`) that fail under jsdom's `/@fs/` virtual FS resolver, causing all 78 UI tests in `tests/ui/` to fail with:

```
Cannot find module '/@fs/workspaces/openclaw-projects/.local/openclaw-gateway/test/setup.ts'
```

UI tests don't need gateway plugin registration — the setup is only needed for gateway integration tests.

## Acceptance Criteria

- [ ] UI tests (`tests/ui/**`) pass when `.local/openclaw-gateway` exists
- [ ] Gateway tests still pass (gateway setup still loads for non-jsdom tests)
- [ ] No changes to `.local/openclaw-gateway/` (gitignored, not our code)

## Approach

Create a thin wrapper `tests/setup-gateway.ts` that checks `typeof window === 'undefined'` before importing the gateway setup. Reference the wrapper from `vitest.config.ts` instead of the gateway setup directly.

--- a/tests/setup-gateway.ts
+++ b/tests/setup-gateway.ts
@@ -1,0 +1,5 @@
+// Only load gateway setup in Node (not jsdom) — gateway imports use
+// relative paths that fail under Vite's /@fs/ resolver in jsdom.
+if (typeof window === 'undefined') {
+  await import('../.local/openclaw-gateway/test/setup.ts');
+}

--- a/vitest.config.ts
+++ b/vitest.config.ts
@@ -62,7 +62,7 @@ export default defineConfig({
     setupFiles: [
       './tests/setup-api.ts',
       './tests/setup-ui.ts',
-      ...(hasGateway ? ['./.local/openclaw-gateway/test/setup.ts'] : []),
+      ...(hasGateway ? ['./tests/setup-gateway.ts'] : []),
     ],
   },
   resolve: {

